templates/InstallChocolatey.ps1.erb

```diff
@@ -38,9 +38,19 @@ param (
   [string]$file
  )
   Write-Output "Downloading $url to $file"
-  $downloader = new-object System.Net.WebClient
+  $downloader = New-Object System.Net.WebCLient
+<% if @proxy_server -%>
+  $proxy = New-Object System.Net.WebProxy("<% @proxy_server -%>:<% @proxy_port -%>", $true)
+  <%- if @proxy_username -%>
+  $passwd = ConvertTo-SecureString "<% @proxy_password -%>" -AsPlainText -Force; ## Website credentials
+  $proxy.Credentials = New-Object System.Management.Automation.PSCredential ("<% @proxy_username -%>", $passwd);
+  $downloader.UseDefaultCredentials = $true
+  <%- end -%>
+  $downloader.Proxy = $proxy
```

Comment:
I'm slightly confused about this. I thought that this was for strictly installing chocolatey itself via the puppet module. It looks like the link you sent was setting the proxy within chocolatey for packages to be installed via "choco install" correct? I did take this from https://github.com/chocolatey/choco/wiki/Proxy-Settings-for-Chocolatey and then just put it in the template file with some if statements.